command/root.go

```diff
@@ -363,25 +367,72 @@ func determineEditor(cmd *cobra.Command) (string, error) {
 	return editorCommand, nil
 }
 
-func ExpandAlias(args []string) ([]string, error) {
-	empty := []string{}
+func ExecuteShellAlias(args []string) error {
+	externalCmd := exec.Command(args[0], args[1:]...)
+	externalCmd.Stderr = os.Stderr
+	externalCmd.Stdout = os.Stdout
+	externalCmd.Stdin = os.Stdin
+	preparedCmd := run.PrepareCmd(externalCmd)
+
+	err := preparedCmd.Run()
+	if err != nil {
+		return fmt.Errorf("failed to run external command: %w", err)
+	}
+
+	return nil
+}
+
+// ExpandAlias processes argv to see if it should be rewritten according to a user's aliases. The
+// second return value indicates whether the alias should be executed in a new shell process instead
+// of running gh itself.
+func ExpandAlias(args []string) (expanded []string, isShell bool, err error) {
+	err = nil
+	isShell = false
+	expanded = []string{}
+
 	if len(args) < 2 {
 		// the command is lacking a subcommand
-		return empty, nil
+		return
 	}
 
 	ctx := initContext()
 	cfg, err := ctx.Config()
 	if err != nil {
-		return empty, err
+		return
 	}
 	aliases, err := cfg.Aliases()
 	if err != nil {
-		return empty, err
+		return
 	}
 
 	expansion, ok := aliases.Get(args[1])
 	if ok {
+		if strings.HasPrefix(expansion, "!") {
+			isShell = true
+			expanded = []string{"sh", "-c", expansion[1:]}
+			if runtime.GOOS == "windows" {
+				// Need to use absolute path for sh on windows
+				shPath, lookErr := exec.LookPath("sh")
+				if lookErr != nil {
+					gitPath, lookErr := exec.LookPath("git")
+					if lookErr != nil {
+						// TODO this error could be better probably
+						err = fmt.Errorf("unable to find sh. you will not be able to use shell aliases on this platform.")
+						return
+					}
+					shPath = filepath.Join(filepath.Dir(gitPath), "..", "bin", "sh.exe")
```

Comment:
This assumes that if we found `git.exe`, we also have access to `sh.exe`, but that might not be true for someone who installed git by other means, perhaps via a package manager. Could we additionally check that `sh.exe` exists before trying to invoke it? This would be mainly to generate nicer error messages